src/nilrt-snac

```diff
@@ -1,127 +1,3 @@
 #!/bin/bash
 # SPDX-License-Identifier: MIT
-set -euo pipefail
-
-
-## CONSTANTS
-SCRIPT_ROOT=$(realpath $(dirname ${BASH_SOURCE}))
-
-## IMPORTS
-source "${SCRIPT_ROOT}/../lib/nilrt-snac/util.sh"
-
-## FUNCTIONS
-print_version() {
-	cat <<-EOF
-	nilrt-snac 0.1
-	Copyright (C) 2024 NI (Emerson Electric)
-	License MIT: MIT License <https://spdx.org/licenses/MIT.html>
-	This is free software: you are free to change and redistribute it.
-	There is NO WARRANTY, to the extent permitted by law.
-	EOF
-}
-
-
-## CLI PARSING
-usage() {
-	cat <<EOF
-$(basename ${BASH_SOURCE}) [--help] [--version] [--yes] <config>
-
-NI LinuxRT Secured, Network-Attached Controller utility.
-
-# Options
--h, --help
-	Print this usage information and exit.
--V, --version
-	Print this script's version information and exit.
--y, --yes
-	Skip user prompts (answer 'yes' to all prompts).
-
-# Commands
-config, configure  Affect the SNAC configuration on this machine.
-
-# Returns
-$EX_OK    when the operation was successful.
-$EX_ERROR    when an unexpected error occurred.
-$EX_USAGE    when given invalid arguments.
-$EX_BAD_ENVIRONEMNT  when the execution environment is invalid.
-$EX_CHECK_FAILURE  when a configuration check fails.
-EOF
-}
-
-
-positionals=()
-yes=false
-
-while [ $# -ge 1 ]; do case "$1" in
-	-h|--help)
-		usage
-		exit 0
-		;;
-	-V|--version)
-		print_version
-		exit 0
-		;;
-	-y|--yes)
-		yes=true
-		shift
-		;;
-	-*|--*)
-		log ERROR "Invalid or unknown option \"$1\"."
-		exit 2
-		;;
-	*)
-		positionals+=($1)
-		shift
-		;;
-esac; done
-
-# If no comand is given, print usage and exit 2.
-if [ ${#positionals[@]} -lt 1 ]; then\
-	log ERROR "Missing required positional arguments.";
-	usage
-	exit 2
-fi
-command=${positionals[0]}
-
-
-# Configure the system for SNAC
-# returns: The return code from configure-nilrt-snac; or 0, when the user does not give final consent.
-cmd_configure() {
-	echo WARNING '!! THIS TOOL IS IN-DEVELOPMENT AND APPROPRIATE ONLY FOR DEVELOPER TESTING !!'
-	echo WARNING '!! Running this tool will irreversibly alter the state of your system.    !!'
-	echo WARNING '!! If you are accessing your system using WiFi, you will lose connection. !!'
-	
-	# prompt for consent
-	local consent=false
-	if $yes; then
-		consent=y
-	else
-		read -p "Do you want to continue with SNAC configuration? [y/N] " consent
-	fi
-
-	case "$consent" in
-		[yY]*)
-			bash "${SCRIPT_ROOT}/../lib/nilrt-snac/configure-nilrt-snac"
-
-			echo ""  # Blank line
-			echo "!! A reboot is now required to affect your system configuration. !!"
-			echo "!! Login with user 'root' and no password.                       !!"
-
-			return $?
-			;;
-		*)
-			return 0
-			;;
-	esac
-}
-
-case "$command" in
-	config|configure)
-		cmd_configure
-		exit $?
-		;;
-	*)
-		log ERROR Unrecognized command \"${command}\"
-		exit $EX_USAGE
-		;;
-esac
+PYTHONPATH=${PYTHONPATH}:/usr/local/lib/nilrt-snac python3 -m nilrt_snac $*
```

Comment:
This wrapper script only works for installations that are using `/usr/local` as their prefix. There are a few was to fix this.

1) You (the project designer) can punt on making a wrapper script entirely and leave it up to the distribution vendor (in this case, meta-nilrt). The vendor has knowledge of where the package is being installed and can write their own wrapper.

This effectively punts the burden to distro maintainers. But it means that you won't necessarily have a nice wrapper script for devs who are building locally.

2) You can use a templating variable in place of the prefix like

```bash
export PYTHONPATH=%prefix%/lib/nilrt-snac
python3 -m nilrt_snac $@
```

And then use `sed` or another templating engine during `make` to write the Make `$(prefix)` value into this file.

This gives you a valid wrapper script for local builds of the file, but requires more logic in the Makefile. For larger projects that use Automake and already have workflows to `./configure` and templatize content, it isn't a big deal. But it would mean new steps for ours.

3) You can assert that the wrapper script should always be installed to the same `prefix` as the other project content, and so the prefix can be assumed within a relative path.

```bash
export PYTHONPATH=$(realpath ../lib/nilrt-snac)
python3 -m nilrt_snac $@
```

This should always work, assuming that downstream installers are using our installation logic. But you might argue that if they are customizing the installation logic, it's OK to make them write their own wrapper.

Reply:
Changed to use realpath, but it is more complicated that what is shown here since realpath only works with cwd.